src/ipred.rs

```diff
@@ -979,33 +993,28 @@ unsafe fn ipred_z2_rust<BD: BitDepth>(
         };
         if filter_strength_0 != 0 {
             filter_edge::<BD>(
-                &mut *topleft.offset(-height as isize),
+                &mut edge,
+                topleft - height as usize,
```

Comment:
If `height` is unsigned, can we cast to `usize` at the beginning of the `fn`?  Same for `width` (and get rid of the `width.try_into().unwrap()`).

Reply:
`width` and `height` are both unsigned I think, but they're used heavily as `c_int`, so casting them to `usize` at the start of the function is going to require that we cast them back to `c_int` in a bunch of places, which I think is just going to make the code noisier.